tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/AuthorityTests.cs

```diff
@@ -113,6 +115,61 @@ public async Task AuthorityValidationTestWithFalseValidateAuthorityAsync()
                 .ConfigureAwait(false);
         }
 
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public async Task UncommonAuthorityValidationTestAsync(bool validateAuthority)
+        {
+            LabResponse labResponse = await LabUserHelper.GetDefaultUserAsync().ConfigureAwait(false);
+            LabUser user = labResponse.User;
+
+            IPublicClientApplication pca = PublicClientApplicationBuilder
+                .Create(labResponse.App.AppId)
+                .WithInstanceDiscoevryEndpoint(false)
+                .WithAuthority("https://bogus.microsoft.com/common", validateAuthority)
+                .WithTestLogging()
+                .Build();
+
+            Trace.WriteLine("Acquire a token using a not so common authority alias");
+
+            HttpRequestException exception = await AssertException.TaskThrowsAsync<HttpRequestException>(() =>
+                 pca.AcquireTokenByUsernamePassword(
+                    s_scopes,
+                     user.Upn,
+                     user.GetOrFetchPassword())
+                     .ExecuteAsync())
+                .ConfigureAwait(false);
+
+            Assert.AreEqual("No such host is known.", exception.Message);
+            Assert.AreEqual(11001,((SocketException)exception.InnerException).ErrorCode);
```

Comment:
I think what this test is trying to say is that authority validation is not performed and MSAL will fail when reaching out to the /token endpoint. This is fine, but I don't see why we'd have an expensive integration test for this, when a unit test can work just as well. We are not interested in testing the integration point between MSAL and an unknwon authority, we're just interested in making sure instance discovery is not done (which by the way you don't assert here).